### PR TITLE
fix: intermittent failure in feature_llmq_simplepose.py

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
kudos to kwvg to report issue: he pointed out that functional tests randomly fail lately.
Bisect pointed out an exact commit: bitcoin#20027 - mockable time everywhere

## What was done?
Added call `mn.node.mockscheduler` as it is done in 20027 for other functional tests

## How Has This Been Tested?
Run 20 times. Without this patch 20% failures; with this patch - zero failures.
```
test/functional/test_runner.py -j20 feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py feature_llmq_simplepose.py
```

## Breaking Changes
N/a

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by knst** [comment]
@kwvg claims that with this patch it still fails randomly on his machine, but only 1/3 of failures compare to without patch.

```
<with patch>
feature_llmq_simplepose.py | � Passed  | 376 s
feature_llmq_simplepose.py | � Passed  | 379 s
feature_llmq_simplepose.py | � Passed  | 382 s
feature_llmq_simplepose.py | � Passed  | 383 s
feature_llmq_simplepose.py | � Passed  | 384 s
feature_llmq_simplepose.py | � Passed  | 384 s
feature_llmq_simplepose.py | � Passed  | 386 s
feature_llmq_simplepose.py | � Passed  | 387 s
feature_llmq_simplepose.py | � Passed  | 387 s
feature_llmq_simplepose.py | � Passed  | 388 s
feature_llmq_simplepose.py | � Passed  | 388 s
feature_llmq_simplepose.py | � Passed  | 389 s
feature_llmq_simplepose.py | � Passed  | 389 s
feature_llmq_simplepose.py | � Passed  | 390 s
feature_llmq_simplepose.py | � Passed  | 390 s
feature_llmq_simplepose.py | � Passed  | 391 s
feature_llmq_simplepose.py | � Passed  | 391 s
feature_llmq_simplepose.py | � Failed  | 69 s
feature_llmq_simplepose.py | � Failed  | 69 s
feature_llmq_simplepose.py | � Failed  | 71 s
```
vs
```
<without patch>
feature_llmq_simplepose.py | � Passed  | 353 s
feature_llmq_simplepose.py | � Passed  | 356 s
feature_llmq_simplepose.py | � Passed  | 351 s
feature_llmq_simplepose.py | � Passed  | 352 s
feature_llmq_simplepose.py | � Passed  | 354 s
feature_llmq_simplepose.py | � Passed  | 350 s
feature_llmq_simplepose.py | � Passed  | 353 s
feature_llmq_simplepose.py | � Passed  | 357 s
feature_llmq_simplepose.py | � Passed  | 346 s
feature_llmq_simplepose.py | � Passed  | 356 s
feature_llmq_simplepose.py | � Passed  | 346 s
feature_llmq_simplepose.py | � Failed  | 62 s
feature_llmq_simplepose.py | � Failed  | 63 s
feature_llmq_simplepose.py | � Failed  | 59 s
feature_llmq_simplepose.py | � Failed  | 61 s
feature_llmq_simplepose.py | � Failed  | 63 s
feature_llmq_simplepose.py | � Failed  | 59 s
feature_llmq_simplepose.py | � Failed  | 61 s
feature_llmq_simplepose.py | � Failed  | 60 s
feature_llmq_simplepose.py | � Failed  | 60 s
```

any ideas how we can improve this simplepose test futhermore?

File: test/functional/feature_llmq_simplepose.py [hunk 1]
```diff
@@ -199,7 +199,8 @@ def repair_masternodes(self, restart):
                     self.start_masternode(mn)
                 else:
                     mn.node.setnetworkactive(True)
-            self.connect_nodes(mn.node.index, 0)
+            self.connect_nodes(mn.node.index, 1)
```
[the remark this answers is not in the document]
Comment (knst) [reply test/functional/feature_llmq_simplepose.py:202]: can't explain, but for me it was a "reliable" fix if I change both node number + call mock-scheduler. Only mockscheduler or only node changes didn't helped me

**Comment by knst** [comment]
> Simply waiting for connections did not help so I tried moving block generation out of the loop and suddenly https://github.com/dashpay/dash/commit/d89f41caeec33b9a4957c95b1a747e26e2e915b9 fixed the issue for me
almost fixed for me.

@kwvg , can you test it, please?
```
feature_llmq_simplepose.py | ✓ Passed  | 355 s
feature_llmq_simplepose.py | ✓ Passed  | 355 s
feature_llmq_simplepose.py | ✓ Passed  | 358 s
feature_llmq_simplepose.py | ✓ Passed  | 359 s
feature_llmq_simplepose.py | ✓ Passed  | 360 s
feature_llmq_simplepose.py | ✓ Passed  | 360 s
feature_llmq_simplepose.py | ✓ Passed  | 361 s
feature_llmq_simplepose.py | ✓ Passed  | 361 s
feature_llmq_simplepose.py | ✓ Passed  | 362 s
feature_llmq_simplepose.py | ✓ Passed  | 366 s
feature_llmq_simplepose.py | ✓ Passed  | 366 s
feature_llmq_simplepose.py | ✓ Passed  | 367 s
feature_llmq_simplepose.py | ✓ Passed  | 367 s
feature_llmq_simplepose.py | ✓ Passed  | 368 s
feature_llmq_simplepose.py | ✓ Passed  | 369 s
feature_llmq_simplepose.py | ✓ Passed  | 369 s
feature_llmq_simplepose.py | ✓ Passed  | 370 s
feature_llmq_simplepose.py | ✓ Passed  | 370 s
feature_llmq_simplepose.py | ✓ Passed  | 371 s
feature_llmq_simplepose.py | ✖ Failed  | 126 s
```

**Comment by kwvg** [comment]
```
dash@228219c1225b:/src/dash$ ./test/functional/test_runner.py -j20 feature_llmq_simplepose.py [...]
[...]
TEST                       | STATUS    | DURATION

feature_llmq_simplepose.py | � Passed  | 381 s
feature_llmq_simplepose.py | � Passed  | 385 s
feature_llmq_simplepose.py | � Passed  | 386 s
feature_llmq_simplepose.py | � Passed  | 386 s
feature_llmq_simplepose.py | � Passed  | 387 s
feature_llmq_simplepose.py | � Passed  | 387 s
feature_llmq_simplepose.py | � Passed  | 388 s
feature_llmq_simplepose.py | � Passed  | 389 s
feature_llmq_simplepose.py | � Passed  | 390 s
feature_llmq_simplepose.py | � Passed  | 392 s
feature_llmq_simplepose.py | � Passed  | 392 s
feature_llmq_simplepose.py | � Passed  | 393 s
feature_llmq_simplepose.py | � Passed  | 393 s
feature_llmq_simplepose.py | � Passed  | 394 s
feature_llmq_simplepose.py | � Passed  | 394 s
feature_llmq_simplepose.py | � Passed  | 395 s
feature_llmq_simplepose.py | � Passed  | 395 s
feature_llmq_simplepose.py | � Passed  | 396 s
feature_llmq_simplepose.py | � Passed  | 396 s
feature_llmq_simplepose.py | � Passed  | 397 s

ALL                        | � Passed  | 7816 s (accumulated)
Runtime: 397 s
dash@228219c1225b:/src/dash$ git rev-parse HEAD
d89f41caeec33b9a4957c95b1a747e26e2e915b9
```

**Comment by UdjinM6** [comment]
I think I figured it out. Previously isolated nodes are trying to fetch blocks they missed as soon as they have a new connection open. When we bump mocktime 10+ minutes in the middle of that we trigger `Timeout downloading block` event https://github.com/dashpay/dash/blob/develop/src/net_processing.cpp#L5171. So these nodes drop "slow" peers (node0, the only connection they have) and everything falls apart. Bumping mocktime only when nodes are synced already should solve the issue. Pls check f8581d89975f78ebde1a66de2bd98788c4422e18.

**Comment by kwvg** [comment]
```
dash@228219c1225b:/src/dash$ ./test/functional/test_runner.py -j20 feature_llmq_simplepose.py [...]
[...]
TEST                       | STATUS    | DURATION

feature_llmq_simplepose.py | � Passed  | 374 s
feature_llmq_simplepose.py | � Passed  | 379 s
feature_llmq_simplepose.py | � Passed  | 384 s
feature_llmq_simplepose.py | � Passed  | 385 s
feature_llmq_simplepose.py | � Passed  | 386 s
feature_llmq_simplepose.py | � Passed  | 387 s
feature_llmq_simplepose.py | � Passed  | 389 s
feature_llmq_simplepose.py | � Passed  | 389 s
feature_llmq_simplepose.py | � Passed  | 390 s
feature_llmq_simplepose.py | � Passed  | 390 s
feature_llmq_simplepose.py | � Passed  | 391 s
feature_llmq_simplepose.py | � Passed  | 393 s
feature_llmq_simplepose.py | � Passed  | 394 s
feature_llmq_simplepose.py | � Passed  | 394 s
feature_llmq_simplepose.py | � Passed  | 395 s
feature_llmq_simplepose.py | � Passed  | 396 s
feature_llmq_simplepose.py | � Passed  | 396 s
feature_llmq_simplepose.py | � Passed  | 397 s
feature_llmq_simplepose.py | � Passed  | 397 s
feature_llmq_simplepose.py | � Passed  | 398 s

ALL                        | � Passed  | 7804 s (accumulated)
Runtime: 398 s

dash@228219c1225b:/src/dash$ git rev-parse HEAD
f8581d89975f78ebde1a66de2bd98788c4422e18
```

**Comment by UdjinM6** [comment]
> Pls check [f8581d8](https://github.com/dashpay/dash/commit/f8581d89975f78ebde1a66de2bd98788c4422e18)

ping @knst

**Review by knst — COMMENTED** [review]
> Pls check [f8581d8](https://github.com/dashpay/dash/commit/f8581d89975f78ebde1a66de2bd98788c4422e18)

this one works perfectly for me.

**Review by UdjinM6 — APPROVED** [review]
ACK :)

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**